torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -466,6 +466,18 @@ RegisterOperators reg({
             };
           }
         }),
+    // TODO schema
+    Operator(prim::Fork, [](Node* stack) -> Operation {
+      return [=](Stack& stack) -> int {
+        throw detail::NewCoroutine();
```

Comment:
This won't throw an exception in the real implementation (control in the caller always continues) normally after a fork. Here is where the task will be added to the scheduler, the task hooked up to a future, and the future returned.

Reply:
For the first commit, I think we should have it just run the sub-block, but still allocate the future (setting ready=true).

Reply:
Also, exceptions are really slow, so it would be great if we could avoid using them in the normal path.